### DISCUSSION TITLE
Shakker.ai’s IP Adapter (style transfer)

### DISCUSSION THREAD

**Discussion by evelryu** [lllyasviel/stable-diffusion-webui-forge#2321] [Ideas]
Can this be implemented as a controlnet feature or other kind of implementation?

https://www.reddit.com/r/StableDiffusion/comments/1gr0d6w/just_tried_shakkerais_ip_adapter_customizing/

**Comment by AnabelleKM** [comment]
Style transfer is already possible by using ControlNet. Example for SDXL:

	ControlNet:		IP-ADAPTER
	Preprocessor:		CLIP-ViT-bigG
	Model:			ip-adapter_xl [4209e9f7]

	Control weight: 	1	(How aggressive you want the style transfer to show up in your image. Adjust to your liking)
	Timestep Range:		0-1	(0 % to 100 %. / Start of image to end of image generation: At what percentage of image generation the style transfer should should take ffect. Adjust to your liking)


1)
On this page is a download link to the file CLIP-Vit-bigG.safetensors
(Be sure to name it like that when downloading)
https://github.com/cubiq/ComfyUI_IPAdapter_plus/blob/main/README.md
**Download to folder \webui\models\ControlNetPreprocessor**

2)
You also need this file "ip-adapter_sdxl.safetensors"
(https://huggingface.co/h94/IP-Adapter/resolve/main/sdxl_models/ip-adapter_sdxl.safetensors).
**Download to folder \webui\models\ControlNet**

3)
Start / Restart Forge to see it in the list of Preprocessors.
